docs/building-rpm-ostree-and-autosd.md

```diff
@@ -0,0 +1,171 @@
+
+## Building rpm-ostree rpm on CentOS Stream9
+
+Create a **ContainerFile** with the content below.
+
+```bash
+FROM centos:stream9
+
+RUN dnf update -y
+RUN dnf install -y 'dnf-command(config-manager)' epel-release
+RUN dnf config-manager --set-enabled crb
+RUN dnf config-manager --add-repo https://buildlogs.centos.org/9-stream/automotive/aarch64/packages-main/
+RUN dnf config-manager --add-repo https://buildlogs.centos.org/9-stream/autosd/aarch64/packages-main/
+RUN if [ $(uname -m) == "aarch64" ]; then dnf install -y --nogpgcheck autosig-u-boot; fi
+RUN dnf install -y --allowerasing --nogpgcheck mock cpio qemu-kvm rpm-build autoconf \
+  automake bison dracut git gpgme-devel gtk-doc libattr-devel libcap-devel libtool \
+  make openssl-devel fuse libarchive libselinux libsoup zlib \
+  gobject-introspection-devel e2fsprogs-devel fuse-devel libarchive-devel \
+  libcurl-devel libsoup-devel systemd-devel xz-devel flex python3-devel swig bc \
+  centos-release-automotive audit-libs-devel bzip2-devel cryptsetup-devel dbus-devel \
+  elfutils-devel firewalld-filesystem gcc-c++ gettext gnu-efi gnu-efi-devel \
+  gnutls-devel gperf hostname kmod-devel libacl-devel libfdisk-devel libgcrypt-devel \
+  libidn2-devel libmicrohttpd-devel libseccomp-devel libxkbcommon-devel \
+  libzstd-devel lz4 lz4-devel pam-devel perl tree valgrind-devel meson \
+  bash-completion tpm2-tss-devel attr dtc
+
+# Install deps required for recent rpm-ostree
+RUN cd rpmbuild/SPECS/ && curl -OL https://gitlab.com/redhat/centos-stream/rpms/rpm-ostree/-/raw/c9s/rpm-ostree.spec
+RUN cd /tmp && dnf builddep -y rpm-ostree.spec
+
+# Lets use the latest from community (see next curl rustup step).
+# Also avoids "cargo metadata: error: failed to run rustc to learn about target-specific information"
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+```
+
+## Building the image
+
+```bash
+podman build -t rpmostreedevel -f ContainerFile
+```
+
+## Running the image
+
+The strategy now is run the container sharing the current **HOME/rpmbuild** dir, so as soon the rpm is created in the container it will be located in the host machine in the running user dir.
+
+```bash
+sudo podman run --rm --privileged -v /home/$USER/rpmbuild/:/home/$USER/rpmbuild/ -ti rpmostreedevel /bin/bash
+
+[root@d1ec4ed543ec /]#
+```
+
+## Building a forked rpm-ostree
+
+Now time to clone your [rpm-ostree](https://github.com/coreos/rpm-ostree) fork and build it.
+
+Cloning the repo and switch to your work branch
+
+```bash
+cd /tmp
+git clone https://github.com/dougsland/rpm-ostree.git && cd rpm-ostree
+git checkout remotes/origin/initoverlayfs -b initoverlayfs
+```
+
+Adding required submodules for the build
+
+```bash
+git submodule update --init
+```
+
+Generating source code
+
+```bash
+cd packaging
+cargo install cargo-vendor-filterer --version ^0.5
```

Comment:
This cargo install should probably be just an extra "RUN" command in the Containerfile